tests/main/snap-update-ns/task.yaml

```diff
@@ -8,21 +8,89 @@ details: |
+    /usr/lib/snapd/snap-update-ns $PLUG_SNAP 2>snap-update-ns.log | MATCH '^$'
+
+    # Connect the plug to the slot.
+    #
+    # NOTE: when the feature is complete the need to use update-ns manually
+    # will go away and we can instead observe the side-effects.
+    snap connect $PLUG_SNAP:shared-content-plug $SLOT_SNAP:shared-content-slot
+
+    # Run the update tool manually.
+    #
+    # NOTE: Currently the tool doesn't make any changes but prints what should
+    # be done. Once this is possible observe the results instead.
+    echo "Check that snap-update-ns does mount, on 1st call, after connecting the content interface"
+    /usr/lib/snapd/snap-update-ns $PLUG_SNAP 2>snap-update-ns.log | MATCH "mount \(/snap/$SLOT_SNAP/x1 /snap/$PLUG_SNAP/x1/import none bind,ro 0 0\)"
+
+    # Check that desired and current profile are the same.
+    diff -Nu /var/lib/snapd/mount/snap.$PLUG_SNAP.fstab /run/snapd/ns/snap.$PLUG_SNAP.fstab
+
+    # Run the update tool again to see that it is idempotent.
+    echo "Check that snap-update-ns does nothing, on 2nd call, after connecting the content interface" 
+    /usr/lib/snapd/snap-update-ns $PLUG_SNAP 2>snap-update-ns.log | MATCH '^$'
+
+    # Check that desired and current profile are (still) the same.
+    diff -Nu /var/lib/snapd/mount/snap.$PLUG_SNAP.fstab /run/snapd/ns/snap.$PLUG_SNAP.fstab
+
+    # Disconnect the plug from the slot so that we can test the other way.
+    snap disconnect $PLUG_SNAP:shared-content-plug $SLOT_SNAP:shared-content-slot
+
+    # NOTE: Currently the tool doesn't make any changes but prints what should
+    # be done. Once this is possible observe the results instead.
+    echo "Check that snap-update-ns does unmount, on 1st call, after disconnecting the content interface" 
+    /usr/lib/snapd/snap-update-ns $PLUG_SNAP 2>snap-update-ns.log | MATCH "unmount \(/snap/$SLOT_SNAP/x1 /snap/$PLUG_SNAP/x1/import none bind,ro 0 0\)"
+
+    # Check that desired and current profile are (still) the same.
+    diff -uN /var/lib/snapd/mount/snap.$PLUG_SNAP.fstab /run/snapd/ns/snap.$PLUG_SNAP.fstab
+
+    # Run the update tool again to see that it is idempotent.
+    echo "Check that snap-update-ns does nothing, on 2nd call, after disconnecting the content interface" 
+    /usr/lib/snapd/snap-update-ns $PLUG_SNAP 2>snap-update-ns.log | MATCH '^$'
+
+    # Check that desired and current profile are (still) the same.
+    diff -uN /var/lib/snapd/mount/snap.$PLUG_SNAP.fstab /run/snapd/ns/snap.$PLUG_SNAP.fstab
+
+    # Discard the namespace so that update has nothing useful to do. Let's just
+    # ensure that it really does use the namespace and doesn't meddle with the
+    # current mount namespace.
+    /usr/lib/snapd/snap-discard-ns $PLUG_SNAP
+    echo "Check that snap-update-ns fails after discarding the mount namespace"
+    /usr/lib/snapd/snap-update-ns $PLUG_SNAP 2>snap-update-ns.log | MATCH "cannot update snap namespace: cannot switch mount namespace: invalid argument"
```

Comment:
I have way more coming. This works but I have also the full-blown version that does everything automatically and I'll be just adding more tests onw.